internal/handlers/boot_artifacts.go

```diff
@@ -0,0 +1,99 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"regexp"
+
+	"github.com/openshift/assisted-image-service/pkg/imagestore"
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	log "github.com/sirupsen/logrus"
+)
+
+type BootArtifactsHandler struct {
+	ImageStore imagestore.ImageStore
+}
+
+var _ http.Handler = &BootArtifactsHandler{}
+
+var bootpathRegexp = regexp.MustCompile(`^/boot-artifacts/(.+)`)
+
+func parseArtifact(path string) (string, error) {
+	match := bootpathRegexp.FindStringSubmatch(path)
+	if len(match) < 1 {
+		return "", fmt.Errorf("malformed download path: %s", path)
+	}
+
+	var artifact string
+	switch match[1] {
+	case "rootfs":
+		artifact = "rootfs.img"
+	case "kernel":
+		artifact = "vmlinuz"
+	default:
+		return "", fmt.Errorf("malformed download path: %s", path)
+	}
+	return artifact, nil
+}
+
+func (b *BootArtifactsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		log.Error("Only GET method is supported with this endpoint.")
+		http.NotFound(w, r)
+		return
+	}
+
+	artifact, err := parseArtifact(r.URL.Path)
+	if err != nil {
+		log.Errorf("failed to parse artifact: %v\n", err)
+		http.NotFound(w, r)
+		return
+	}
+	version, arch, err := b.parseQueryParams(r.URL.Query())
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		_, err = w.Write([]byte(err.Error()))
+		if err != nil {
+			log.Errorf("Failed to write response: %v\n", err)
+		}
+		return
+	}
+
+	isoFileName := b.ImageStore.PathForParams(imagestore.ImageTypeFull, version, arch)
+	fileReader, err := isoeditor.GetFileFromISO(isoFileName, fmt.Sprintf("/images/pxeboot/%s", artifact))
```

Comment:
"/images/pxeboot" already appears several times in the repo, consider defining it as a const

Reply:
This is the only occurrence that isn't in tests.

Specifically most uses are mocking or asserting on grub config. I'm not sure it's worth doing string interpolation for all of those.